transport.go

```diff
@@ -320,9 +319,7 @@ func (t *Transport) doDial(
 		logger,
 		version,
 	)
-	t.connMx.Lock()
 	t.handlers[srcConnID] = conn
-	t.connMx.Unlock()
```

Comment:
Why is the mutex not needed here anymore?

Reply:
This could potentially be a problem: `newClientConn` does a lot of work initialization all the connection-related structs.

I'm not sure how big of a problem this actually is though, so I'm ok with merging this PR for now. We might need to change this is someone shows up with a mutex profile indicating contention of this mutex.